### ISSUE TITLE
time_zone inside date_histogram composite didnt work ?

### ISSUE THREAD

**Issue by viyancs** [elastic/elasticsearch#49357]
elasticsearch 7.2 here my mapping
````{
    "reports": {
        "mappings": {
            "properties": {
                "browser_name": {
                    "type": "keyword"
                },
                "browser_version": {
                    "type": "keyword"
                },
                "city": {
                    "type": "keyword"
                },
                "clicks": {
                    "type": "integer"
                },
                "country": {
                    "type": "keyword"
                },
                "creative_id": {
                    "type": "integer"
                },
                "device_brand": {
                    "type": "keyword"
                },
                "device_model": {
                    "type": "keyword"
                },
                "device_type": {
                    "type": "keyword"
                },
                "impressions": {
                    "type": "integer"
                },
                "interval_start": {
                    "type": "date",
                    "format": "yyyy-MM-dd HH:mm:ss"
                },
                "os_name": {
                    "type": "keyword"
                },
                "os_version": {
                    "type": "keyword"
                },
                "region": {
                    "type": "keyword"
                },
                "requests": {
                    "type": "integer"
                },
                "tags": {
                    "type": "keyword"
                },
                "zone_id": {
                    "type": "integer"
                }
            }
        }
    }
}
````

then i try rerquest using like this
````
{
    "query": {
        "bool": {
            "must": [
                {
                    "range": {
                        "interval_start": {
                            "gte": "2019-11-17 17:00:00",
                            "lte": "2019-11-19 16:59:59"
                        }
                    }
                }
            ],
            "filter": [
                {
                    "terms": {
                        "creative_id": [
                            "3787"
                        ]
                    }
                }
            ]
        }
    },
    "aggs": {
        "histograms2": {
            "date_histogram": {
                "field": "interval_start",
                "interval": "day",
                "format": "yyyy-MM-dd",
                "time_zone": "Asia/Jakarta"
            }
        },
        "statistic": {
            "composite": {
                "size": 1000,
                "sources": [
                    {
                        "date_f": {
                            "date_histogram": {
                                "field": "interval_start",
                                "format": "yyyy-MM-dd",
                                "interval": "day",
                                "time_zone": "Asia/Jakarta"
                            }
                        }
                    },
                    {
                        "country": {
                            "terms": {
                                "field": "country"
                            }
                        }
                    },
                    {
                        "region": {
                            "terms": {
                                "field": "region"
                            }
                        }
                    },
                    {
                        "city": {
                            "terms": {
                                "field": "city"
                            }
                        }
                    },
                    {
                        "device_type": {
                            "terms": {
                                "field": "device_type"
                            }
                        }
                    },
                    {
                        "device_brand": {
                            "terms": {
                                "field": "device_brand"
                            }
                        }
                    },
                    {
                        "device_model": {
                            "terms": {
                                "field": "device_model"
                            }
                        }
                    },
                    {
                        "os_name": {
                            "terms": {
                                "field": "os_name"
                            }
                        }
                    },
                    {
                        "browser_name": {
                            "terms": {
                                "field": "browser_name"
                            }
                        }
                    }
                ]
            },
            "aggs": {
                "sum_impressions": {
                    "sum": {
                        "field": "impressions"
                    }
                },
                "sum_requests": {
                    "sum": {
                        "field": "requests"
                    }
                },
                "sum_clicks": {
                    "sum": {
                        "field": "clicks"
                    }
                }
            }
        }
    }
}
````
the result like this
----------
````
{
    "took": 1782,
    "timed_out": false,
    "_shards": {
        "total": 10,
        "successful": 10,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 10000,
            "relation": "gte"
        },
        "max_score": null,
        "hits": []
    },
    "aggregations": {
        "histograms2": {
            "buckets": [
                {
                    "key_as_string": "2019-11-18",
                    "key": 1574010000000,
                    "doc_count": 1992566
                },
                {
                    "key_as_string": "2019-11-19",
                    "key": 1574096400000,
                    "doc_count": 2440622
                }
            ]
        },
        "statistic": {
            "after_key": {
                "date_f": "2019-11-17",
                "country": "Australia",
                "region": "Australian Capital Territory",
                "city": "Canberra",
                "device_type": "Mobile",
                "device_brand": "Apple",
                "device_model": "iPhone",
                "os_name": "iOS",
                "browser_name": "Chrome"
            },
            "buckets": [
                {
                    "key": {
                        "date_f": "2019-11-17",
                        "country": "",
                        "region": "",
                        "city": "",
                        "device_type": "Desktop",
                        "device_brand": "-",
                        "device_model": "-",
                        "os_name": "Linux",
                        "browser_name": "Chrome"
                    },
                    "doc_count": 1,
                    "sum_impressions": {
                        "value": 1.0
                    },
                    "sum_requests": {
                        "value": 1.0
                    },
                    "sum_clicks": {
                        "value": 0.0
                    }
                }
````

----------

from that result date_histogram outside composite have correct result the date formatted to time zone "Asia/Jakarta", but from result date_histogram inside composite the date didn't formatted there is still have "2019-11-17" , this is bug ?

**Comment by salvatore-campagna** [comment]
@viyancs Do you have a simpler reproduction case including data?

**Comment by imotov** [comment]
I think we can close this one. Here is the reproduction:
```
DELETE test
PUT test
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date",
        "format": "yyyy-MM-dd HH:mm:ss"
      }
    }
  }
}

POST test/_bulk?refresh
{"index": {}}
{"@timestamp": "2022-01-01 00:00:00" }
{"index": {}}
{"@timestamp": "2022-01-01 01:00:00"}
{"index": {}}
{"@timestamp": "2022-01-01 22:00:00" }
{"index": {}}
{"@timestamp": "2022-01-01 23:00:00" }


POST test/_search
{
  "size": 0,
  "aggs": {
    "stand_alone": {
      "date_histogram": {
        "field": "@timestamp",
        "format": "yyyy-MM-dd",
        "interval": "day",
        "time_zone": "Asia/Jakarta"
      }
    },
    "composite": {
      "composite": {
        "sources": [
          {
            "inside_composite": {
              "date_histogram": {
                "field": "@timestamp",
                "format": "yyyy-MM-dd",
                "interval": "day",
                "time_zone": "Asia/Jakarta"
              }
            }
          }
        ]
      }
    }
  }
}
```
In 7.2.0 I get:
```
  "aggregations" : {
    "composite" : {
      "after_key" : {
        "inside_composite" : "2022-01-01"
      },
      "buckets" : [
        {
          "key" : {
            "inside_composite" : "2021-12-31"
          },
          "doc_count" : 2
        },
        {
          "key" : {
            "inside_composite" : "2022-01-01"
          },
          "doc_count" : 2
        }
      ]
    },
    "stand_alone" : {
      "buckets" : [
        {
          "key_as_string" : "2022-01-01",
          "key" : 1640970000000,
          "doc_count" : 2
        },
        {
          "key_as_string" : "2022-01-02",
          "key" : 1641056400000,
          "doc_count" : 2
        }
      ]
    }
  }
```
While in 7.14.0 I get:
```
  "aggregations" : {
    "composite" : {
      "after_key" : {
        "inside_composite" : "2022-01-02"
      },
      "buckets" : [
        {
          "key" : {
            "inside_composite" : "2022-01-01"
          },
          "doc_count" : 2
        },
        {
          "key" : {
            "inside_composite" : "2022-01-02"
          },
          "doc_count" : 2
        }
      ]
    },
    "stand_alone" : {
      "buckets" : [
        {
          "key_as_string" : "2022-01-01",
          "key" : 1640970000000,
          "doc_count" : 2
        },
        {
          "key_as_string" : "2022-01-02",
          "key" : 1641056400000,
          "doc_count" : 2
        }
      ]
    }
  }
```

I think we fixed it somewhere between 7.2 and 7.14.

**Comment by viyancs** [comment]
yea thanks for the release, last time I fixed using java formatter, seem in the new version the bug is gone
